workers/pipeline/update_pipeline.go

```diff
@@ -79,9 +85,45 @@ func updatePipeline(p *gaia.Pipeline) error {
 		// Install gem forcefully.
 		cmd := exec.Command(path, "install", "-f", pipelineCopyPath)
 		if out, err := cmd.CombinedOutput(); err != nil {
-			gaia.Cfg.Logger.Error("error", string(out[:]))
+			return fmt.Errorf("cannot install ruby gem: %s", string(out[:]))
+		}
+	case gaia.PTypeNodeJS:
```

Comment:
At some point we'll need to refactor this into something sensible like a plugin type factory of some sorts, because this switch case is getting ridicules. :D

Reply:
Yeah, I completely agree. 

I personally think for now it's "fine" since we have two or three switches which you need to modify when you add a new language so that's overall not much but for the future, a lot of refactoring will be required!

But that is a problem for the future Michel (😅)

Reply:
:D :D :D Fine. :D